### PR TITLE
Using new shared include files in FPGA ANR design

### PR DESCRIPTION
**Pull request by tyoungsc**
# Existing Sample Changes
## Description

Deleting files from the FPGA ANR design in favour of the new shared include files. Adding rom_base.hpp to the include directory and updating the README accordingly. Also fixed the spelling of "lambda".

## Type of change
- [X] code cleanup

## How Has This Been Tested?
- [X] Command Line
- [X] regtest

### CODE REVIEW

File: DirectProgramming/DPC++FPGA/ReferenceDesigns/anr/src/anr.hpp [hunk 1]
```diff
@@ -18,9 +18,9 @@
 #include "anr_params.hpp"
 #include "column_stencil.hpp"
 #include "constants.hpp"
+#include "constexpr_math.hpp"
```
Comment (mdbtucker) [inline DirectProgramming/DPC++FPGA/ReferenceDesigns/anr/src/anr.hpp:21]: Add comment about location of this file

File: DirectProgramming/DPC++FPGA/ReferenceDesigns/anr/src/column_stencil.hpp [hunk 2]
```diff
@@ -4,13 +4,13 @@
 #include <CL/sycl.hpp>
 #include <sycl/ext/intel/fpga_extensions.hpp>
 
+#include "constexpr_math.hpp"
```
Comment (mdbtucker) [inline DirectProgramming/DPC++FPGA/ReferenceDesigns/anr/src/column_stencil.hpp:7]: file location comment

File: DirectProgramming/DPC++FPGA/ReferenceDesigns/anr/src/column_stencil.hpp [hunk 3]
```diff
@@ -4,13 +4,13 @@
...
 
+#include "constexpr_math.hpp"
 #include "data_bundle.hpp"
-#include "mp_math.hpp"
 #include "shift_reg.hpp"
 #include "unrolled_loop.hpp"
 
 using namespace sycl;
-using namespace hldutils;
+using namespace fpga_tools;
```
Comment (mdbtucker) [inline DirectProgramming/DPC++FPGA/ReferenceDesigns/anr/src/column_stencil.hpp:13]: Do we want to use 'using' statements like this?  I thought this went against best coding practices?

You do this in other files as well, so if you decide to get rid of it here please grep for 'using namespace' in your code and fix it everywhere.

File: DirectProgramming/DPC++FPGA/ReferenceDesigns/anr/src/constants.hpp [hunk 4]
```diff
@@ -2,7 +2,7 @@
 #define __CONSTANTS_HPP__
 
 #include <CL/sycl/INTEL/ac_types/ac_int.hpp>
-#include "mp_math.hpp"
+#include "constexpr_math.hpp"
```
Comment (mdbtucker) [inline DirectProgramming/DPC++FPGA/ReferenceDesigns/anr/src/constants.hpp:5]: file location comment

File: DirectProgramming/DPC++FPGA/ReferenceDesigns/anr/src/main.cpp [hunk 5]
```diff
@@ -14,12 +14,12 @@
 #include "anr.hpp"
 #include "anr_params.hpp"
 #include "constants.hpp"
+#include "constexpr_math.hpp"
```
Comment (mdbtucker) [inline DirectProgramming/DPC++FPGA/ReferenceDesigns/anr/src/main.cpp:17]: file location comment

Comment (mdbtucker) [inline DirectProgramming/DPC++FPGA/ReferenceDesigns/anr/src/main.cpp:17]: Is this header actually used?  I see no other changes in this file, I would have thought you'd have to update the namespace if any functions from this file were actually used here.

File: DirectProgramming/DPC++FPGA/ReferenceDesigns/anr/src/qfp.hpp [hunk 6]
```diff
@@ -6,7 +6,7 @@
 
 #include <CL/sycl/INTEL/ac_types/ac_int.hpp>
 
-#include "mp_math.hpp"
+#include "constexpr_math.hpp"
```
Comment (mdbtucker) [inline DirectProgramming/DPC++FPGA/ReferenceDesigns/anr/src/qfp.hpp:9]: file location

Comment (mdbtucker) [inline DirectProgramming/DPC++FPGA/ReferenceDesigns/anr/src/qfp.hpp:9]: Is this header really used?  I don't see namespace updates in the file.

File: DirectProgramming/DPC++FPGA/ReferenceDesigns/anr/src/qfp_exp_lut.hpp [hunk 7]
```diff
@@ -1,6 +1,7 @@
 #ifndef __QFP_EXP_LUT_HPP__
 #define __QFP_EXP_LUT_HPP__
 
+#include "constexpr_math.hpp"
```
Comment (mdbtucker) [inline DirectProgramming/DPC++FPGA/ReferenceDesigns/anr/src/qfp_exp_lut.hpp:4]: file location

File: DirectProgramming/DPC++FPGA/ReferenceDesigns/anr/src/row_stencil.hpp [hunk 8]
```diff
@@ -5,13 +5,13 @@
 #include <sycl/ext/intel/fpga_extensions.hpp>
 #include <limits>
 
+#include "constexpr_math.hpp"
```
Comment (mdbtucker) [inline DirectProgramming/DPC++FPGA/ReferenceDesigns/anr/src/row_stencil.hpp:8]: file location

**Review by mdbtucker — COMMENTED** [review]
Do you need to update the README file to indicate the new location for the shared include files?

File: DirectProgramming/DPC++FPGA/ReferenceDesigns/anr/README.md [hunk 9]
```diff
@@ -191,14 +191,11 @@ The following source files can be found in the `src/` sub-directory.
 |`data_bundle.hpp`                | A generic library for bundling data to move between kernels; essentially an array.
 |`dma_kernels.hpp`                | Contains kernels that move data between the host and device, as well as reading/writing data between the FPGA and device memory.
 |`intensity_sigma_lut.hpp`        | A RAM LUT for the intensity sigma values.
-|`mp_math.hpp`                    | Metaprogramming math helper functions.
```
Comment (mdbtucker) [inline DirectProgramming/DPC++FPGA/ReferenceDesigns/anr/README.md:194]: I don't think common include files should be removed entirely from the README - we should just point out that they live in a separate directory.  I believe that's what we have done for other samples that use the common files.

Comment (tyoungsc) [reply DirectProgramming/DPC++FPGA/ReferenceDesigns/anr/README.md:194]: Addressed in most recent commit.
I put a little sentence after that says "for this, this, and this, file see INCLUDE DIRECTORY".

**Review by mdbtucker — APPROVED** [review]

**Pull request merged**